### ISSUE TITLE
Null checking fails when using MemberNotNullWhen and property patterns

### ISSUE THREAD

**Issue by skarllot** [dotnet/roslyn#49750]
**Diagnostic ID**: CS8600 and CS8603

### Describe the bug

The compiler fails to check nullability using `MemberNotNullWhenAttribute` and property patterns.

### Steps To Reproduce

The following code triggers the compiler warning.

```csharp
#nullable enable

public struct Test
{
    [MemberNotNullWhen(true, nameof(Value))]
    public bool IsOk { get; }
    public bool IsError { get; }
    public string? Value { get; }
}

public class Program {
    public string Main(Test test) {
        switch (test){
            case { IsOk: true }:
                return test.Value;    // <-- Possible null reference return
            case { IsError: true }:
                return String.Empty;
            default:
                return String.Empty;
        }
    }
}
```

### Expected behavior

Follows the attributes and check correctly.

## Additional context

The code above does not trigger the compiler warning.

```csharp
public class Program {
    public string Main(Test test) {
        switch (test){
            case { } when test.IsOk:
                return test.Value;    // <-- No error here
            case { } when test.IsError:
                return String.Empty;
            default:
                return String.Empty;
        }
    }
}
```

The example code can be found on Sharplab: [link](https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4AEBMBGAsAKBQBgAIUMA6AEQEsIBzAOwHtZKBjKUgYQYBM4BBOhAA2ATyiUoAbly4AxHQCuQoRGBC4hOILVwZeAMyFYCBSxiEAKnFi4A3rkKPCAbQCycMMEQA5BjG9KQgDqABZaABQwJnAANISCYHAMAGbhAGrCCnAAlNkAug5OKIbADAxChACSUADyANaEtoQ0cDCShAC+hY7FhKXlVVAAoggIDAiNza3tXThOxIYk+AD8hBlCWZMtbZ24s7i96MQYAOyN3QvHRFawGOE35vCw2edz805QAO6UMCwhhJFrDBsvY3u93ixoBomtV6iBCFFNh0QBdweCUGcnjBSOssu0nAB6AmEAA8AFoyYQAApMcQ6eKBQgIODJRBaFgaDGotGQqDQwYjMYIeGIjTI7loopnADKUUodBopCGYAADjARNIwZLCLxkhAlDAUVrtVLCLKEPLFcq1RqJbstbN5hdDhhrkC0PcgQigS9QZKvj8/gCsSC7fNefyOoRPmE6N7YKRYXUjSb0ZigTjMnB8Y4iaSKYRfJpRuNCGFmWGnBHJlGY1p49jqoLxinU/MMWa5Qqlar1Zq27r9UJDZW052Ld3rX27Y6nLPHM7Fq7LED9J7YA3fXaA79/oDnn7U9Wmri4PDGOZFMpOq22z10wnTznCHnyZSi4ghWW2aPCMfOtGsYNomwwlsKv7tjKXZWr2trGmig4Gred4dualo9ja/ZovO9r7DguAJNYKoQByxwUNQ9BMDArDsFwvACMIYgSHYFzOHwMBysACjwAAqlAtBwOE7GcdxcAWBACDbOw7gwCEPCEEghDCRaXHwOJkmtOwVJjCqiDqnElR0OWPxwNwhAALyEHqQh8nEfDKAwnyuAalAquoFkItE+QXPK8AIIIFR8sIpnEGghDuJ4Ph+AEyihFoymUKpGjwglSUXIe6IlGUFQAEqtAo/mnls0z2naLr4M4eThR4XgIFAxU7Hhkq9BFtW+P4gRxXQqWieE/QVBip5xEshCJJFCDZHaGXanlMAFXQRWWYNWZYSarWIPVll0HAnyVZMY21Z0q3vE1aItTVUUdbFsY9fAfXZcQJxDYQxEIBAYD1Use0HRtk3wY402SrN82LY9T6/utdUeT9dXHfMOGzB0QA==)
